web/viewer.js

```diff
@@ -1323,15 +1328,16 @@ var PDFViewerApplication = {
     this.updateScaleControls = true;
   },
 
-  rotatePages: function pdfViewRotatePages(delta) {
+  rotatePages: function pdfViewRotatePages(delta, noRendering) {
     var pageNumber = this.page;
     this.pageRotation = (this.pageRotation + 360 + delta) % 360;
     this.pdfViewer.pagesRotation = this.pageRotation;
     this.pdfThumbnailViewer.pagesRotation = this.pageRotation;
 
-    this.forceRendering();
-
-    this.pdfViewer.scrollPageIntoView(pageNumber);
+    if (!noRendering) {
```

Comment:
I like to avoid double negatives when possible. How about we make a setRotation(delta) function that only updates values and then change rotatePages to call setRotation and the code in this 'if' statement. Then in setInitialView you just have to call setRotation(...)